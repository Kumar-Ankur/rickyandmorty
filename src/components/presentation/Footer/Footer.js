import React from "react";
import "./Footer.scss";
import * as constanst from "../../../constants";

const Footer = () => {
  return (
    <div className="footer_container" data-test='component-footer'>
      <h2 className="footer_heading">
        {constanst.FOOTER_HEADING} <br />
        <br />
        <span style={{ position: 'absolute', top: '2rem'}}>{constanst.FOOTER_SUB_HEADING}{" "}</span>
        <br />
        <a href="http://kumarankur.in/" target="_blank" rel="noopener noreferrer" className="footer_link">
          {constanst.FOOTER_LINK}
        </a>
      </h2>
    </div>
  );
};

export default Footer;
