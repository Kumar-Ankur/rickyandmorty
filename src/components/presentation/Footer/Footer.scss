.footer_container {
    width: 100vw;
    height: 20vh;
    background: black;
    display: flex;
    justify-content: center;
    position: relative;
    bottom: -7rem;
  }
  
  .footer_heading {
    color: white;
    font-family: "AccoladeTOTMed W01 Regular";
    position: relative;
    top: 35px;
    font-size: 1.2rem;
  }

  .footer_link {
      color: gold;
      margin-left: 2rem;
      text-transform: uppercase;
      font-weight: normal;
  }
  
  @media only screen and (max-width: 949px) and (min-width: 550px),
    only screen and (max-width: 549px) and (min-width: 400px),
    only screen and (max-width: 399px) and (min-width: 350px),
    only screen and (max-width: 349px) {
    .footer_heading {
    //   font-size: 1rem;
    //   top: 25px;
    }
  }