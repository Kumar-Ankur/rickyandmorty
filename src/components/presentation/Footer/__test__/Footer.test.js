import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../Footer';

describe('check footer component', () => {
    test('generate snapshot for footer component', () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper).toMatchSnapshot();
    });

    test('check footer component render without crash', () => {
        const wrapper = shallow(<Footer />);
        const footerComponent = wrapper.find(`[data-test='component-footer']`);
        expect(footerComponent.length).toBe(1);
    });
})