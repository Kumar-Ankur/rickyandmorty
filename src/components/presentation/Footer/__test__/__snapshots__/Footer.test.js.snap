// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check footer component generate snapshot for footer component 1`] = `ShallowWrapper {}`;
